### ISSUE TITLE
If you use the requirejs. can't use proj4js. 

### ISSUE THREAD

**Issue by realrnjs** [openlayers/ol3#3990]
proj4 is not defined

if (ol.ENABLE_PROJ4JS && typeof proj4 == 'function')

This condition is always false.

**Comment by ahocevar** [comment]
Fixed in v3.8.

**Comment by afernandezp** [comment]
I don't think it is fixed in v.3.8. Currently I'm working with OL3 V3.8.2 and I'm still getting that exception: "proj4 is not defined"

**Comment by ahocevar** [comment]
You're right. The issue fixed with #3910 was for libraries that are baked into the library. Sorry for the confusion.

**Comment by juristr** [comment]
Wouldn't it be enough to include it in the UMD at the top?? 

Something like

```javascript
(function(root, factory) {
  if (typeof exports === "object") {
    module.exports = factory(require('proj4');
  } else if (typeof define === "function" && define.amd) {
    define(['proj4'], factory);
  } else {
    root.ol = factory(root.proj4);
  }
}(this, function(proj4) {
   ...
});
```

**Comment by ahocevar** [comment]
At the top of what?

**Comment by juristr** [comment]
@ahocevar of the final build file that is being produced. Just download the latest release and check the `dist/ol.js` file.

**Comment by ahocevar** [comment]
This should work, yes. Pull request welcome!

**Comment by juristr** [comment]
@ahocevar Ok, I'm testing this out today on my local configuration and if everything works properly, I'll submit a PR.

**Comment by fredj** [comment]
Fixed with #4609
